### PR TITLE
Add a derive_constructor option to generate struct constructors

### PR DESCRIPTION
**Pull request by staktrace**

### CODE REVIEW

File: src/bindgen/ir/structure.rs [hunk 1]
```diff
@@ -232,6 +232,47 @@ impl Source for Struct {
...
+                && !self.fields.is_empty()
+            {
+                if !wrote_start_newline {
+                    wrote_start_newline = true;
+                    out.new_line();
+                }
+
+                out.new_line();
+
+                let arg_renamer = RenameRule::GeckoCase;
```
Comment (eqrion) [inline src/bindgen/ir/structure.rs]: Could we use `config.function.rename_args` here? Just so it's customizable.

**Comment by staktrace** [comment]
Actually it might still be useful, if I add another option to add IPDL-friendly goop to the structs (which would consist of a private no-arg constructor and a friend declaration for the ParamTraits). I'll try that at some point.

File: src/bindgen/ir/structure.rs (hunk 1 above)
Comment (staktrace) [reply src/bindgen/ir/structure.rs]: Updated patch to pull out a lambda that uses the config.function.rename_args if it exists.

**Comment by eqrion** [comment]
Thanks!

**Pull request merged**